### PR TITLE
add dependabot.yml to check the component version upgrades automatically

### PR DESCRIPTION
**Pull request by liweinan**
@jamezp I copied the `dependabot.yml` from RESTEasy as initial version of the file here. wdyt?

### CODE REVIEW

**Comment by liweinan** [comment]
After this work it done at `resteasy-spring`, I'll do the setup for `resteasy-spring-boot`.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    directory: "/"
+    schedule:
+      interval: "daily"
+    ignore:
+      - dependency-name: 'jakarta.el:jakarta.el-api'
+        update-types: ['version-update:semver-major', 'version-update:semver-minor']
+      - dependency-name: 'org.glassfish:jakarta.el'
+        update-types: ['version-update:semver-major', 'version-update:semver-minor']
+      - dependency-name: 'jakarta.persistence:jakarta.persistence-api'
+        update-types: ['version-update:semver-major', 'version-update:semver-minor']
```
Comment (jamezp) [inline .github/dependabot.yml:18]: I don't see a dependency on any of these in the POM's.

**Review by jamezp — CHANGES_REQUESTED** [review]
There is a chance you need to review the dependencies in the POM's to ensure we ignore any that we don't want to use the latest for. I didn't see any at first glance, but we might just want to make sure.

**Comment by liweinan** [comment]
@jamezp Thanks for checking! I'll modify this file according to `resteasy-spring` usage.

**Comment by liweinan** [comment]
@jamezp I have removed the unnecessary ignores sections. I guess this is a basic setup now, and if there are more requirements I can tweak it further, wdyt?

**Review by jamezp — APPROVED** [review]
Looks good to me. If there are things we don't want to upgrade in the future we can add it to the ignore list.

**Comment by liweinan** [comment]
@jamezp yes sir!

**Pull request merged**